inst/pages/20_beta_diversity.qmd

```diff
@@ -226,6 +226,86 @@ plotReducedDim(tse, "Unifrac",
+
+There has been a long-lasting controversy surrounding the use of rarefaction in
+microbial ecology. The main concern is that rarefaction would omit data. 
+However, if the subsampling process is repeated a sufficient number of times, 
+and if the rarefaction depth is set to the lowest sequence count found across all samples, no data will be omitted. 
+Moreover, Patrick Schloss has demonstrated that rarefaction is "the only method that could control for variation in uneven sequencing effort when measuring commonly used alpha and beta diversity metrics" [@Schloss2023].
+
+The function `vegan::avgdist` uses rarefaction to compute beta diversity :
+
+```{r, fig.cap = "MDS plot based on the Bray-Curtis distances using rarefaction on the GlobalPatterns dataset."}
+
+tse <- runMDS(tse, 
+              assay.type="counts",
+              ntop = nrow(GlobalPatterns),
+              FUN = avgdist,
+              sample = min(colSums(assay(tse, "counts"))), # rarefaction depth
+			        distfun = vegan::vegdist, # Use vegdist distance function
+			        dmethod = "bray", # Bray-Curtis dissimilarity (vegdist argument)
+			        iterations = 10 # 10 iterations should be enough for distance calculations
+)
+
+plotReducedDim(tse, "MDS")
+```
+
+`ntop` is a `runMDS` option. Here it is set to total number of features in `GlobalPatterns` dataset so that they are not filtered.
+
+The argument `sample` is by default set to the smallest sequence count across 
+all samples. This ensures that no sample will be removed during the rarefaction 
+process.
+
+The argument `iterations` is by default set to 100. 10 iterations should be 
+sufficient for distance calculations.
+
+To use transformations after the rarefaction of the samples and before the beta diversity calculation, `vegan::avgdist` has the argument `transf` :
+
+```{r, fig.cap = "MDS plot based on the Bray-Curtis distances using rarefaction with clr transformation on the GlobalPatterns dataset."}
+# Define custom transformation function..
+clr <- function (x) {
+  vegan::decostand(x, method="clr", pseudocount=1)
+}
+
+# Run transformation after rarefactions before calculating the distances..
+tse <- runMDS(tse, assay.type="counts",
+                           ntop=nrow(GlobalPatterns), 
+                           FUN = vegan::avgdist, # Custom distance that includes rarefaction
+                           sample=min(colSums(assay(tse, "counts"))), # rarefaction depth 
+			                     distfun=vegdist,  # Use vegdist distance function
+                           dmethod = "euclidean", # euclidean dissimilarity
+			                     iterations = 10, 
+			                     transf=clr) 
+
+plotReducedDim(tse, "MDS")
+```
+
+The mia package has the rarefaction function `subsampleCounts` that randomly
+subsamples one of the assays of a TreeSummarizedExperiment dataset. However, the 
```

Comment:
TreeSummarizedExperiment -> _TreeSummarizedExperiment_ (always in italics, can you also check elsewhere)